src/main/java/quickcache/model/flashcard/Difficulty.java

```diff
@@ -0,0 +1,73 @@
+package quickcache.model.flashcard;
+
+import static java.util.Objects.requireNonNull;
+
+import quickcache.commons.util.AppUtil;
+
+public class Difficulty {
+
+    public static final String MESSAGE_CONSTRAINTS = "Difficulty names should only be "
+            + "low, medium, or high. "
+            + "No spaces allowed. Cannot be empty.";
+    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
+
+    public final String difficulty;
+
+    /**
+     * Constructs a {@code Difficulty}.
+     *
+     * @param difficulty A valid difficulty.
+     */
+    public Difficulty(String difficulty) {
+        requireNonNull(difficulty);
+        AppUtil.checkArgument(isValidDifficultyName(difficulty), MESSAGE_CONSTRAINTS);
+        this.difficulty = difficulty.toUpperCase();
+    }
+
+    /**
+     * Constructs a {@code Difficulty}.
+      */
+    public Difficulty() {
+        this.difficulty = "UNSPECIFIED";
+    }
+
+    /**
+     * Returns true if a given string is a valid tag name.
+     */
+    public static boolean isValidDifficultyName(String test) {
+
+        if (test.matches(VALIDATION_REGEX)) {
```

Comment:
Don't think we need `VALIDATION_REGEX` if we are already checking it against `Difficulties.values()`

Reply:
I have done the changes, thanks!